phoenix-core-client/src/main/java/org/apache/phoenix/compile/CreateTableCompiler.java

```diff
@@ -209,6 +250,151 @@ public MutationPlan compile(CreateTableStatement create) throws SQLException {
+            List<PTable> legitimateSiblingViewList =
+                    ViewUtil.findAllDescendantViews(childLinkTable, config, parentTenantIdInBytes,
+                            parentSchemaNameInBytes, parentToBe.getTableName().getBytes(),
+                            HConstants.LATEST_TIMESTAMP, true).getFirst();
+            if (legitimateSiblingViewList.size() > 0) {
+                PTable siblingView = legitimateSiblingViewList.get(0);
+                Expression siblingViewWhere = getWhereFromView(connection, siblingView);
+                Set<PColumn> siblingViewPkColsInWhere = new HashSet<>();
+                if (siblingViewWhere != null) {
+                    ViewWhereExpressionValidatorVisitor siblingViewValidatorVisitor =
+                            new ViewWhereExpressionValidatorVisitor(parentToBe,
+                                    siblingViewPkColsInWhere, new HashSet<>());
+                    siblingViewWhere.accept(siblingViewValidatorVisitor);
+                }
+                if (!pkColumnsInWhere.equals(siblingViewPkColsInWhere)) {
+                    LOGGER.info("Setting the view type as READ_ONLY because the set of PK columns" +
+                            " are different from its sibling views'");
+                    return ViewType.READ_ONLY;
+                }
+            }
+        }
+        return ViewType.UPDATABLE;
+    }
+
+    /**
+     * Get the where Expression of given view.
+     * @param connection The client connection
+     * @param view PTable of the view
+     * @return A where Expression
+     * @throws SQLException
+     */
+    private Expression getWhereFromView(final PhoenixConnection connection, final PTable view)
+            throws SQLException {
+        String viewStatement = view.getViewStatement();
+        if (viewStatement == null) {
+            return null;
+        }
+        SelectStatement select = new SQLParser(viewStatement).parseQuery();
+        ColumnResolver resolver = FromCompiler.getResolverForQuery(select, connection);
+        StatementContext context = new StatementContext(new PhoenixStatement(connection), resolver);
+        BitSet isViewColumnReferencedToBe = new BitSet(view.getColumns().size());
+        ExpressionCompiler expressionCompiler = new ColumnTrackingExpressionCompiler(context,
+                isViewColumnReferencedToBe);
+        ParseNode whereNode = select.getWhere();
+        return whereNode.accept(expressionCompiler);
+    }
+
+    /**
+     * Check if the primary key columns are in order (consecutive in position) as they are
+     * defined, providing their positions list.
+     * @param pkPositions A positions list of PK columns to be checked
+     * @param tablePkPositions The positions list of the table's PK columns
+     * @param isMultiTenant Whether the parent table is multi tenant
+     * @return
+     */
+    private boolean isPkColumnsInOrder(final List<Integer> pkPositions,
```

Comment:
Can you make sure, this holds for SALTED tables.

Reply:
The column position will move one position back, just like the multi-tenant table